src/devices/philips_AJ7010.c

```diff
@@ -0,0 +1,164 @@
+/* 
+ * Philips outdoor temperature sensor -- used with various Philips clock 
+ * radios (tested on AJ7010).
+ * This is inspired from the other Philips driver made by Chris Coffey.
+ *
+ * A complete message is 43 bits:
+ *      4-bit initial preamble, always 0
+ *      4-bit packet separator, always 0
+ *      Packets are repeated 3 times.
+ * 
+ * 43-bit data packet format:
+ *
+ * 00001000 10100100 01000101 00101001 110 : g_philips_21.1_ch2_B.cu8
+ * 00001011 01001111 10000101 00100001 111 : g_philips_21.4_ch1_C.cu8
+ * 00001011 01000000 10100100 11001111 001 : gph_bootCh1_17.cu8
+ * 00001000 10100011 11000100 11001111 101 : gph_bootCh2_17.cu8
+ * 00000110 11011100 01100100 10111110 000 : gph_bootCh3_17.cu8
+ *
+ * Data format is:
+ * 0000cccc ccc????? ???????ttt tttttttt ttt
+ *
+ * c - channel: 0x5A=channel 1, 0x45=channel 2, 0x36=channel 3 (7 bits)
+ * t - temperature in ADC value that is then converted to deg. C. (14 bits)
+ * s - CRC: non-standard CRC-?, poly 0x?, init 0x?
+ *  
+ * Copyright (C) 2018 Nicolas Jourden <nicolas.jourden@laposte.net>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+#include "decoder.h"
+
+#define PHILIPS_BITLEN       43
+#define PHILIPS_DATALEN      5
+#define PHILIPS_PACKETLEN    6
+#define PHILIPS_STARTNIBBLE  0x0
+
+// We have those values from the reading of the probe:
+#define PHILIPS_MATH_COEF_XB      14576.0
+#define PHILIPS_MATH_COEF_YB      36.8
+#define PHILIPS_MATH_COEF_XA      1440.0
+#define PHILIPS_MATH_COEF_YA      -14.3
+#define PHILIPS_MATH_OFFSET      -20.0
+// TODO: look for a more efficient method:
+#define PHILIPS_MATH_EXPR(v1)    (( ( (PHILIPS_MATH_COEF_YB-PHILIPS_MATH_COEF_YA)/(PHILIPS_MATH_COEF_XB-PHILIPS_MATH_COEF_XA) ) * v1 ) + PHILIPS_MATH_OFFSET)
+
+static int philips_AJ7010_callback(r_device *decoder, bitbuffer_t *bitbuffer) 
+{
+    uint8_t packet[PHILIPS_DATALEN];
+    data_t *data;
+    unsigned int i = 0;
+    unsigned int j = 0;
+    uint8_t c_crc = 0;
+    uint8_t r_crc = 0;
+    uint8_t channel = 0;
+    float temperature = 0;
+    uint32_t tmp = 0;
+
+    // Invert the data:
+    bitbuffer_invert(bitbuffer); // ~ on data.
+
+    // Correct number of rows?
+    if (bitbuffer->num_rows != 1) {
+        if (decoder->verbose > 1) {
+            fprintf(stderr, "%s: wrong number of rows (%d)\n",
+                    __func__, bitbuffer->num_rows);
+        }
+        return 0;
```

Comment:
For the return codes you could look at this:
https://github.com/merbanan/rtl_433/blob/6b6b89321b0cca261819a6694dec21f30995b82c/include/r_device.h#L24-L33

I guess this would result in DECODE_ABORT_LENGTH or DECODE_ABORT_EARLY